### PR TITLE
[RPC] Readme for RPC samples

### PR DESCRIPTION
**Pull request by vaavva**

### CODE REVIEW

File: sdk/samples/core/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ The samples' instructions include specifics for Linux based systems. The Linux e
   sudo apt-get install build-essential curl zip unzip tar pkg-config
   ```
 - Have [Git](https://git-scm.com/download) installed.
-<!-- - Have Microsoft [vcpkg](https://github.com/microsoft/vcpkg) package manager and [Mosquitto MQTT C client](TODO) or [Eclipse Paho MQTT C client](https://www.eclipse.org/paho/) installed. This installation may take an extended amount of time (~15-20 minutes).
+- Have Microsoft [vcpkg](https://github.com/microsoft/vcpkg) package manager and [Mosquitto MQTT C client](https://mosquitto.org/download/) or [Eclipse Paho MQTT C client](https://www.eclipse.org/paho/) installed. This installation may take an extended amount of time (~15-20 minutes).
```
Comment (RLeclair) [inline sdk/samples/core/README.md:28]: Given that we have the automatic vcpkg installation now, would it be useful to add instructions on how to use the auto install?

File: sdk/samples/core/README.md [hunk 2]
```diff
@@ -83,41 +83,34 @@ The samples' instructions include specifics for Linux based systems. The Linux e
...
-      export VCPKG_DEFAULT_TRIPLET=x64-linux
-      <!-- export VCPKG_ROOT=<FULL PATH to vcpkg> -->
-      ```
+## Getting Sarted
 
-  2. Set broker details in the sample:
+### Set up the Sample
+- Open the file of the sample you'd like to run. Ex. [sdk/samples/core/mosquitto_rpc_client_sample.c](https://github.com/Azure/azure-sdk-for-c/blob/feature/v2/sdk/samples/core/mosquitto_rpc_client_sample.c)
+- Fill out the User-defined parameters section at the top of the file - at minimum cert path, key path, and hostname, but all of these are customizable for your solution.
+- In the main function, adjust any configuration options as needed
```
Comment (RLeclair) [inline sdk/samples/core/README.md:113]: Nit:
```suggestion
- In the main function, adjust any configuration options as needed.
```

File: sdk/samples/core/README.md [hunk 3]
```diff
@@ -83,41 +83,34 @@ The samples' instructions include specifics for Linux based systems. The Linux e
...
-      ```bash
-      export VCPKG_DEFAULT_TRIPLET=x64-linux
-      <!-- export VCPKG_ROOT=<FULL PATH to vcpkg> -->
-      ```
+## Getting Sarted
 
-  2. Set broker details in the sample:
+### Set up the Sample
+- Open the file of the sample you'd like to run. Ex. [sdk/samples/core/mosquitto_rpc_client_sample.c](https://github.com/Azure/azure-sdk-for-c/blob/feature/v2/sdk/samples/core/mosquitto_rpc_client_sample.c)
+- Fill out the User-defined parameters section at the top of the file - at minimum cert path, key path, and hostname, but all of these are customizable for your solution.
```
Comment (RLeclair) [inline sdk/samples/core/README.md:112]: Nit: 
```suggestion
- Fill out the user-defined parameters section at the top of the file - at minimum cert path, key path, and hostname, but all of these are customizable for your solution.
```

File: sdk/samples/core/README.md [hunk 4]
```diff
@@ -127,7 +120,10 @@ Set the following environment variables for all samples:
     ```bash
     mkdir build
     cd build
-    cmake -DAZ_MQTT_TRANSPORT_IMPL=MOSQUITTO -DAZ_PLATFORM_IMPL=POSIX ..
+    # for mosquitto samples
+    cmake -DAZ_MQTT_TRANSPORT_IMPL=MOSQUITTO -DAZ_PLATFORM_IMPL=POSIX -DCMAKE_TOOLCHAIN_FILE=<path_to_vcpkg_repo>/scripts/buildsystems/vcpkg.cmake ..
+    # for paho samples
+    cmake -DAZ_MQTT_TRANSPORT_IMPL=PAHO -DAZ_PLATFORM_IMPL=POSIX -DCMAKE_TOOLCHAIN_FILE=<path_to_vcpkg_repo>/scripts/buildsystems/vcpkg.cmake ..
     ```
 2. Compile and run the sample.
```
Comment (RLeclair) [inline sdk/samples/core/README.md:129]: Would be good to mention here that you can edit MQTT configuration default values in `az_mqtt_config.h` (like the port number for example)

**Review by RLeclair — APPROVED** [review]
LGTM! Left some minor nits and comments

File: sdk/samples/core/README.md (hunk 4 above)
Comment (vaavva) [reply sdk/samples/core/README.md:130]: Added in the set up section!

File: sdk/samples/core/README.md (hunk 1 above)
Comment (vaavva) [reply sdk/samples/core/README.md]: Done! Removed explicit vcpkg instructions and added a link to learn more about manually setting these options.

**Pull request merged**